pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -202,6 +204,22 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 		Replica:                 helpers.DetermineReplica(ctx, n.operatorKubeClient, clusterManagerMode, nil),
 		HostedMode:              clusterManager.Spec.DeployOption.Mode == operatorapiv1.InstallModeHosted,
 	}
+
+	// If there are some invalid feature gates of registration, will output condition `InvalidRegistrationFeatureGates` in ClusterManager.
+	if clusterManager.Spec.RegistrationConfiguration != nil && len(clusterManager.Spec.RegistrationConfiguration.FeatureGates) > 0 {
+		featureGateArgs, invalidFeatureGates := helpers.FeatureGatesArgs(clusterManager.Spec.RegistrationConfiguration.FeatureGates, helpers.ComponentHubKey)
+		if len(invalidFeatureGates) == 0 {
+			config.RegistrationFeatureGates = featureGateArgs
```

Comment:
> I notice that with this change we will disable addonmanaged and defaultClusterset feature by default. It will bring some backward compatible problem. How about we keep the original setting if RegistrationConfiguration is nil?

Yes, you are right.  I have fixed.

Reply:
> I think you would want to set the same condition to true when there is no invalid feature gates.
> 
> Otherwise, we will always see false condition if user change the feature gates to correct

ok

Reply:
> nit comment. Other looks good.
> 
> /approve
> 
> I think we should add some integration tests to check whether the condition is returned correctly with invalid features. Not a blocker for this one, but could be in a followup PR,

👌🏻